controllers/managedmcg_controller.go

```diff
@@ -594,3 +610,38 @@ func (r *ManagedMCGReconciler) getCSVByPrefix(name string) (*opv1a1.ClusterServi
 
 	return csv, nil
 }
+
+func (r *ManagedMCGReconciler) ensureConsolePlugin(clusterVersion string) error {
+	// The base path to where the request are sent
+	basePath := console.GetBasePath(clusterVersion)
+
+	// Get mcg console Deployment
+	mcgConsoleDeployment := console.GetDeployment(r.namespace)
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      mcgConsoleDeployment.Name,
+		Namespace: mcgConsoleDeployment.Namespace,
+	}, mcgConsoleDeployment)
+	if err != nil {
+		return err
+	}
+
+	// Create/Update mcg console Service
+	mcgConsoleService := console.GetService(r.ConsolePort, r.namespace)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), r.Client, mcgConsoleService, func() error {
+		return controllerutil.SetControllerReference(mcgConsoleDeployment, mcgConsoleService, r.Scheme)
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
```suggestion
	if err != nil {
```
`CreateOrUpdate` will not error out if the object exists, it'll just proceed to update it in that case.